Formula/opencv.rb

```diff
@@ -84,6 +84,14 @@ def install
       -DPYTHON3_INCLUDE_DIR=#{py3_include}
     ]
 
+    # The compiler on older Mac OS cannot build some OpenCV files using AVX2
+    # extensions, failing with errors such as
+    # "error: use of undeclared identifier '_mm256_cvtps_ph'"
+    # Work around this by not trying to build AVX2 code.
+    if MacOS.version <= :yosemite
+      args << "-DCPU_DISPATCH=SSE4_1,SSE4_2,AVX"
```

Comment:
AVX is not present on all CPUs. Is it not safer to use `-DENABLE_AVX2=OFF` here instead of `CPU_DISPATCH`? (from reading the OpenCV documentation)

Reply:
I tried that first, but it didn't work for me, since `ENABLE_AVX2` only controls the code that is unconditionally built into the library. OpenCV will still (try to) build AVX2 code, even on CPUs that don't support AVX2, in case you try to run the resulting binary on an AVX2-capable machine (it gets activated at run time rather than compile time). Plus, the `ENABLE_*` flags are deprecated anyway.

It *might* be possible that if you try and build OpenCV on Yosemite on a new CPU that does support AVX2 that it will unconditionally build AVX2 code, but I have no way of testing that, and I didn't want to overreach with this patch. I imagine that you'd have few if any users in that situation anyway, as there's no reason to be stuck on a pre-10.12 machine if you have a modern Mac. (And you don't officially support 10.10 anyway.)